### PR TITLE
Workaround broken special APC when running x64 on arm64 windows

### LINKED ISSUE #100425: Runtime crashing on startup on emulated x64-on-ARM64 Windows
Repro steps:
1. Install .net9preview3 X64 SDK from https://aka.ms/dotnet/9.0.1xx/daily/dotnet-sdk-win-x64.exe
2. Modify the environment variable path to x64 "C:\Program Files\dotnet\x64".
3. Create new console app.
`dotnet new console `

Expected Result:
Project can be created successfully.

Actual Result:
Creating project or build fails and will be blocking.
image
image

dotnet --info
.NET SDK:
 Version:           9.0.100-preview.3.24163.23
 Commit:            33549194e2
 Workload version:  9.0.100-manifests.42eb89a3
 MSBuild version:   17.10.0-preview-24162-02+0326fd7c9

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22631
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\x64\sdk\9.0.100-preview.3.24163.23\

.NET workloads installed:
There are no installed workloads to display.

Host:
  Version:      9.0.0-preview.3.24162.24
  Architecture: x64
  Commit:       e4fceb3ae5

.NET SDKs installed:
  9.0.100-preview.3.24163.23 [C:\Program Files\dotnet\x64\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 9.0.0-preview.3.24162.20 [C:\Program Files\dotnet\x64\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 9.0.0-preview.3.24162.24 [C:\Program Files\dotnet\x64\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 9.0.0-preview.3.24162.12 [C:\Program Files\dotnet\x64\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  arm64 [C:\Program Files\dotnet]
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

### PR DESCRIPTION
**Pull request by janvorli**
In ARM64 windows older than 24H2, the special APC is broken when running x64 emulation. The callback that gets invoked doesn't get an argument with correct CONTEXT of the interrupted location. This change disables using the special APC for runtime suspension when running on the affected Windows versions.

Close #100425

### CODE REVIEW

**Comment by janvorli** [comment]
cc: @mangod9, @tommcdon

**Comment by jkotas** [comment]
We have the same code in NAOT: https://github.com/dotnet/runtime/blob/main/src/coreclr/nativeaot/Runtime/windows/PalRedhawkMinWin.cpp#L636-L639 . Could you please make the same fix there as well?

File: src/coreclr/vm/threads.cpp [hunk 1]
```diff
@@ -8130,6 +8131,21 @@ void Thread::InitializeSpecialUserModeApc()
...
 
+#ifdef HOST_AMD64
+    typedef BOOL (WINAPI *IsWow64Process2Proc)(HANDLE hProcess, USHORT *pProcessMachine, USHORT *pNativeMachine);
+
+    IsWow64Process2Proc pfnIsWow64Process2Proc = (IsWow64Process2Proc)GetProcAddress(hKernel32, "IsWow64Process2");
+    USHORT processMachine, hostMachine;
+    if (pfnIsWow64Process2Proc != nullptr &&
+        (*pfnIsWow64Process2Proc)(GetCurrentProcess(), &processMachine, &hostMachine) &&
+        (hostMachine == IMAGE_FILE_MACHINE_ARM64) &&
+        !IsWindowsVersionOrGreater(10, 0, 26100))
```
[the remark this answers is not in the document]
Comment (janvorli) [reply src/coreclr/vm/threads.cpp:8142]: 10, Windows 11 also use version 10 for some reason.

[the remark this answers is not in the document]
Comment (jkotas) [reply src/coreclr/vm/threads.cpp:8142]: Marketing version != build version.

**Review by mangod9 — APPROVED** [review]

**Review by jkotas — APPROVED** [review]
Thanks

**Review by tommcdon — APPROVED** [review]
Thank you!

**Pull request merged**